pyvista/utilities/features.py

```diff
@@ -36,26 +36,34 @@ def voxelize(mesh, density=None, check_surface=True):
 
     >>> import pyvista as pv
     >>> from pyvista import examples
-    >>> mesh = pv.PolyData(examples.load_uniform().points)
-    >>> vox = pv.voxelize(mesh, density=0.5)
-    >>> vox.plot()
+    >>> mesh = examples.download_bunny_coarse().clean()
+    >>> vox = pv.voxelize(mesh, density=0.01)
+    >>> vox.plot(show_edges=True)
 
     Create a voxelized mesh using unequal density dimensions.
 
-    >>> mesh = pv.PolyData(examples.load_uniform().points)
-    >>> vox = pv.voxelize(mesh, density=[0.5, 0.9, 1.4])
-    >>> vox.plot()
+    >>> vox = pv.voxelize(mesh, density=[0.01, 0.005, 0.002])
+    >>> vox.plot(show_edges=True)
 
     """
     if not pyvista.is_pyvista_dataset(mesh):
         mesh = pyvista.wrap(mesh)
     if density is None:
         density = mesh.length / 100
-    if isinstance(density, (int, float)):
+    if isinstance(density, (int, float, np.number)):
         density_x, density_y, density_z = [density] * 3
-    if isinstance(density, (list, set, tuple)):
+    elif isinstance(density, (collections.abc.Sequence, np.ndarray)):
```

Comment:
Thanks for the review @akaszynski and sorry for invalidating it. I had to fix one more thing I noticed late: we used to allow `set` in there. This is dangerous because sets are not sequences (they don't have well-defined order) but can be unpacked, leading to surprises:
```py
>>> x, y, z = {-2, -3, -4}

>>> x, y, z
(-4, -3, -2)
```
And while I was at it I generalized the types a bit, allowing both sequences and arrays as well as numpy scalars.

Reply:
Sharp eye @adeak.

Quite right about sets.